### ISSUE TITLE
Question: Virtual callback with const functions

### ISSUE THREAD

**Issue by jo-m** [bytedeco/javacpp#117]
Hi! I have a question regarding an extended version of the example in the [readme](https://github.com/bytedeco/javacpp#creating-callback-functions). In short, what to do if the (pure) virtual C++ function I want to override from Java is declared const, as in `virtual void bar() const = 0;` ? I cannot override it by using

```
public native @Virtual(true) void bar();
```

The function overriden in Java never gets called from C++ as the type signature is not the same. A complete example is attached, run it with `make run`. The desired output is `Callback in Java`, which you get when you remove `const` from the end of the `bar` declarations in `Foo.h`.
[report.zip](https://github.com/bytedeco/javacpp/files/387869/report.zip)

**Comment by jo-m** [comment]
As this is tagged as `enhancement` now, I assume there is not yet a solution to do this. What would be the best way to achieve this? a `@ConstFn` annotation?

**Comment by saudet** [comment]
I was thinking of adding a value to `@Virtual`, but something general might be wiser yes.. Actually, what about `@Const(function=true)`? Since C++ reuses keywords for this, I try to mimic these kinds of pattern.

**Comment by jo-m** [comment]
How to differ between const return values and const functions? `@Const(function=bool, return=bool)`?

**Comment by saudet** [comment]
There's already a value for that: `@Const(value=true, function=true)`

**Comment by jo-m** [comment]
Ah yes, in this case I think that's the way to go.

**Comment by saudet** [comment]
Actually, looking back at the code, there is already *some* functionality for that. Before the `@Virtual` annotation was added, we already had `FunctionPointer` and in that case there is a peer class for each function, so I made it so that when we put `@Const` on the class, it has the desired effect. When the support for virtual functions was added, for the sake of consistency, I made is so that when we put `@Const` on the enclosing class, all virtual functions become `const`. We can't have one function `const` and another one non `const` though. 

So, I wonder, would this be enough for your case? I have not encountered an example used in the wild where this was not enough yet, but I do acknowledge the limitation...
